File: tests/output/jpa/src/main/javagen/restaurant/jpa_sequence_metamodel/dtos/restaurant/ReservationItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_sequence_metamodel.dtos.restaurant;

import java.time.LocalDateTime;

import jakarta.annotation.Generated;

/**
 * Détail d'une réservation en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface ReservationItem {

	/**
	 * Identifiant de la réservation.
	 */
	Integer getId();

	/**
	 * Date et heure de la réservation.
	 */
	LocalDateTime getDateReservation();

	/**
	 * Nombre de personnes.
	 */
	Integer getNombrePersonnes();

	/**
	 * Indique si la réservation est confirmée.
	 */
	Boolean getConfirmee();

	/**
	 * Client ayant fait la réservation.
	 */
	Integer getClientIdClient();

	/**
	 * Restaurant concerné par la réservation.
	 */
	Integer getRestaurantIdRestaurant();

	/**
	 * Hydrate values of instance.
	 * @param id value to set.
	 * @param dateReservation value to set.
	 * @param nombrePersonnes value to set.
	 * @param confirmee value to set.
	 * @param clientIdClient value to set.
	 * @param restaurantIdRestaurant value to set.
	 */
	void hydrate(Integer id, LocalDateTime dateReservation, Integer nombrePersonnes, Boolean confirmee, Integer clientIdClient, Integer restaurantIdRestaurant);
}

File: tests/output/focus/src/model/restaurant/reservation-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {e, entity, EntityToType} from "@focus4/entities";
import {DO_BOOLEEN, DO_DATE_HEURE, DO_ID, DO_QUANTITE} from "../../domains";

export type ReservationItem = EntityToType<ReservationItemEntityType>;
export type ReservationItemEntityType = typeof ReservationItemEntity;

export const ReservationItemEntity = entity({
    id: e.field(DO_ID, f => f.optional()
        .label("restaurant.reservation.id")
    ),
    dateReservation: e.field(DO_DATE_HEURE, f => f
        .label("restaurant.reservation.dateReservation")
    ),
    nombrePersonnes: e.field(DO_QUANTITE, f => f
        .label("restaurant.reservation.nombrePersonnes")
    ),
    confirmee: e.field(DO_BOOLEEN, f => f.defaultValue(false)
        .label("restaurant.reservation.confirmee")
    ),
    clientIdClient: e.field(DO_ID, f => f
        .label("restaurant.reservation.clientIdClient")
    ),
    restaurantIdRestaurant: e.field(DO_ID, f => f
        .label("restaurant.reservation.restaurantIdRestaurant")
    )
});
